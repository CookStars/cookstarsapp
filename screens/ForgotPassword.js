import { firebase } from '../firebaseconfig'
import React, { Component, Fragment } from 'react'
import { Text, View, StyleSheet, SafeAreaView } from 'react-native'
import { FormInput, FormButton, ErrorMessage } from '../components'
// import ErrorMessage from '../components/ErrorMessage'
import { Formik } from 'formik'
import * as Yup from 'yup'

const validationSchema = Yup.object().shape({
    email: Yup.string()
        .label('Email')
        .email('Enter a valid email')
        .required('Please enter a registered email'),
})

export default class ForgotPassword extends Component {
    constructor(props) {
        super(props)
    }

    handlePasswordReset = ({ email }) => {
        const { navigation } = this.props

        firebase
            .auth()
            .sendPasswordResetEmail(email)
            .then(function (user) {
                console.log(this.props)
                console.log('success email has been sent')
                navigation.navigate('Login')
            })
            .catch(function (e) {
                console.log(e)
            })
    }

    render() {
        return (
            <SafeAreaView style={styles.container}>
                <Text style={styles.text}>Forgot Password?</Text>
                <Formik
                    initialValues={{ email: '' }}
                    onSubmit={(values, actions) => {
                        this.handlePasswordReset(values, actions)
                    }}
                    validationSchema={validationSchema}
                >
                    {({
                        handleChange,
                        values,
                        handleSubmit,
                        errors,
                        isValid,
                        touched,
                        handleBlur,
                        isSubmitting,
                    }) => (
                        <Fragment>
                            <FormInput
                                name="email"
                                value={values.email}
                                onChangeText={handleChange('email')}
                                placeholder="Enter email"
                                autoCapitalize="none"
                                iconName="ios-mail"
                                iconColor="#2C384A"
                                onBlur={handleBlur('email')}
                            />
                            <ErrorMessage
                                errorValue={touched.email && errors.email}
                            />
                            <View style={styles.buttonContainer}>
                                <FormButton
                                    buttonType="outline"
                                    onPress={handleSubmit}
                                    title="Send Email"
                                    backgroundColor={colors.orange}
                                    buttonColor={colors.extra}
                                    disabled={!isValid || isSubmitting}
                                />
                            </View>
                            <ErrorMessage errorValue={errors.general} />
                        </Fragment>
                    )}
                </Formik>
            </SafeAreaView>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        marginTop: 150,
    },
    text: {
        color: '#333',
        fontSize: 24,
        marginLeft: 25,
    },
    buttonContainer: {
        margin: 25,
    },
})
