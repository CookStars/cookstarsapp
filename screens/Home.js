import React, { Component } from "react";
import {
  StyleSheet,
  Text,
  View,
  Button,
  Image,
  ScrollView,
  Dimensions,
  TouchableOpacity,
} from "react-native";
import favicon from "../assets/favicon.png";
import { CardView, Cards } from "../components";
import { recipes } from "../Seed";
import { db } from "../firebaseconfig.js";

const weekdays = [
  "Sunday",
  "Monday",
  "Tuesday",
  "Wednesday",
  "Thursday",
  "Friday",
  "Saturday",
];

export default class HomeScreen extends Component {
  constructor(props) {
    super(props);
    this.state = {
      vegan: [],
      meatlover: [],
      recipeFinished: false
    };
    this.recipeCompleted = this.recipeCompleted.bind(this)
    this.getRecipes = this.getRecipes.bind(this);
  }

  recipeCompleted = () => {
    this.setState({recipeFinished: true});
  };

  async getRecipes(pref) {
    let allRecipes;
    const recipes = await db.collection("recipes").doc(pref).get();
    if (recipes.exists) {
      allRecipes = recipes.data().recipe;
      console.log("Success, recipes found");
    } else {
      console.log("No data found");
    }

    this.setState({ vegan: allRecipes });
  }

  componentDidMount() {
    this.getRecipes("vegan");
  }

  render() {
    const cards = () => {
      return weekdays.map((weekday, index) => (
        <Cards
          key={index}
          day={weekday}
          index={index}
          navigation={this.props.navigation}
          recipes={this.state.vegan}
          userInfo = {this.props.userInfo}
          recipeFinished={this.state.recipeFinished}
          recipeCompleted={this.recipeCompleted}
        />
      ));
    };

    if (!this.state.vegan.length) {
      return null;
    }
    return (
      <View style={styles.container}>
        <CardView
          style={styles.card}
          navigation={this.props.navigation}
          recipes={this.state.vegan}
          userInfo={this.props.userInfo}
          recipeFinished={this.state.recipeFinished}
          recipeCompleted={this.recipeCompleted}
        />
        <Text style={styles.Text}>Recipes of the Week</Text>
        <ScrollView
          vertical={true}
          contentContainerStyle={styles.scrollArea_contentContainerStyle}
        >
          {cards()}
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
  scrollArea_contentContainerStyle: {
    flexWrap: "wrap",
    alignContent: "space-around",
    flexDirection: "row",
    justifyContent: "space-around",
  },
  Text: {
    fontSize: 30,
  },
});
