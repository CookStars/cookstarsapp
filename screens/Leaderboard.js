import React, { useState } from "react";
import { StyleSheet, Text, View, ScrollView } from "react-native";


export default function Leaderboard() {
  const users = [
    {
      firstName: "Grace",
      lastName: "Shopper",
      userName: "GShopper",
      email: "Gshopper@gmail.com",
      points: "50",
      vegan: false,
      favoriteRecipes: {},
      recipeHistory: {},
      imageURL: "",
      key: "1"

    },
    {
      firstName: "Lucy",
      lastName: "Doe",
      userName: "BadCat",
      email: "JDoe@gmail.com",
      points: "24",
      vegan: true,
      favoriteRecipes: {},
      recipeHistory: {},
      imageURL: "",
      key: "2"

    },
    {
      firstName: "Mary",
      lastName: "Doe",
      userName: "Chef123",
      email: "JDoe@gmail.com",
      points: "20",
      vegan: true,
      favoriteRecipes: {},
      recipeHistory: {},
      imageURL: "",
      key: "3"

    },
    {
      firstName: "Carl",
      lastName: "Doe",
      userName: "Foodie123",
      email: "JDoe@gmail.com",
      points: "10",
      vegan: true,
      favoriteRecipes: {},
      recipeHistory: {},
      imageURL: "",
      key: "4"

    },
    {
      firstName: "Suzie",
      lastName: "Doe",
      userName: "GordRamsay",
      email: "JDoe@gmail.com",
      points: "1",
      vegan: true,
      favoriteRecipes: {},
      recipeHistory: {},
      imageURL: "",
      key: "5"

    }
  ]

  return (
    <View style={styles.container}>
      <ScrollView>
        {
          users.map((user) => {
            return (
              <View key={user.key}>

                <Text style={styles.user}>{user.userName} Total Points: {user.points}</Text>
              </View>
            )
          })
        }
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    // alignItems: "center",
    // justifyContent: "center",
  },
  user: {
    marginTop: 24,
    padding: 30,
    backgroundColor: '#f18f01',
    fontSize: 24

  }
});
