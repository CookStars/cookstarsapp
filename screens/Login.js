import React, { Component } from 'react'
import {
    StyleSheet,
    Text,
    View,
    TextInput,
    TouchableHighlight,
    ActivityIndicator,
    Image,
    Alert,
} from 'react-native'
import { firebase } from '../firebaseconfig'
import 'firebase/functions'
import { colors } from '../utils/constants'

export default class Login extends Component {
    state = {
        email: '',
        password: '',
        errorMessage: null,
        loggingIn: false,
    }

    onClickListener = (viewId) => {
        Alert.alert('Alert', 'Button pressed ' + viewId)
        return viewId
    }

    handleLogin = async () => {
        const { email, password } = this.state
        this.setState({ loggingIn: true })

        // Set persistence locally. This will make sure user is logged in through firebase until they log out
        await firebase
            .auth()
            .setPersistence(firebase.auth.Auth.Persistence.LOCAL)
            .then(() => {
                return firebase
                    .auth()
                    .signInWithEmailAndPassword(email, password)
            })
            .catch((error) => {
                // Handle Errors here.
                this.setState({ loggingIn: false })
                this.setState({ errorMessage: error.message })
            })
    }

    render() {
        return (
            <View style={styles.container}>
                <View style={styles.welcomeContainer}>
                    <Image
                        source={require('../assets/cookstarslogo.png')}
                        style={styles.welcomeImage}
                    />
                </View>
                <View style={styles.errorMessage}>
                    {this.state.errorMessage && (
                        <Text style={styles.error}>
                            {this.state.errorMessage}
                        </Text>
                    )}
                </View>

                {/* Email Field */}
                <View style={styles.inputContainer}>
                    <TextInput
                        style={styles.inputs}
                        placeholder="Email"
                        keyboardType="email-address"
                        underlineColorAndroid="transparent"
                        onChangeText={(email) => this.setState({ email })}
                        value={this.state.email}
                        id="emailInput"
                    />
                </View>

                {/* Password Field */}
                <View style={styles.inputContainer}>
                    <TextInput
                        style={styles.inputs}
                        placeholder="Password"
                        secureTextEntry
                        underlineColorAndroid="transparent"
                        onChangeText={(password) => this.setState({ password })}
                        value={this.state.password}
                    />
                </View>

                {/* Login Button */}
                <TouchableHighlight
                    style={[styles.buttonContainer, styles.loginButton]}
                    onPress={() => this.handleLogin()}
                >
                    {this.state.loggingIn ? (
                        <ActivityIndicator size="large"></ActivityIndicator>
                    ) : (
                            <Text style={styles.loginText}>Login</Text>
                        )}
                </TouchableHighlight>

                {/* Forgot password Button */}
                <TouchableHighlight
                    style={styles.buttonContainer}
                    onPress={() =>
                        this.props.navigation.navigate('ForgotPassword')
                    }
                >
                    <Text>Forgot your password?</Text>
                </TouchableHighlight>

                {/* Register Button */}
                <TouchableHighlight
                    style={styles.buttonContainer}
                    onPress={() =>
                        this.props.navigation.navigate('Registration')
                    }
                >
                    <Text>Register</Text>
                </TouchableHighlight>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: colors.offWhite,
    },
    inputContainer: {
        borderBottomColor: colors.offWhite,
        backgroundColor: colors.background,
        borderRadius: 30,
        borderBottomWidth: 1,
        width: 250,
        height: 45,
        marginBottom: 20,
        flexDirection: 'row',
        alignItems: 'center',
    },
    inputs: {
        height: 45,
        marginLeft: 16,
        borderBottomColor: colors.offWhite,
        flex: 1,
    },
    inputIcon: {
        width: 30,
        height: 30,
        marginLeft: 15,
        justifyContent: 'center',
    },
    buttonContainer: {
        height: 45,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        marginBottom: 20,
        width: 250,
        borderRadius: 30,
    },
    loginButton: {
        backgroundColor: colors.orange,
    },
    loginText: {
        color: colors.offWhite,
        fontWeight: 'bold',
    },
    welcomeContainer: {
        alignItems: 'center',
        marginTop: 10,
        marginBottom: 20,
    },
    welcomeImage: {
        width: 300,
        height: 250,
        resizeMode: 'contain',
        marginTop: 3,
        marginLeft: -10,
    },
})
