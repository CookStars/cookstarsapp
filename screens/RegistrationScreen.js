import { firebase } from '../firebaseconfig'

import React, { useState } from 'react'
import {
    Text,
    TextInput,
    TouchableOpacity,
    View,
    StyleSheet,
    ActivityIndicator,
} from 'react-native'
import DropDownPicker from 'react-native-dropdown-picker'
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import { db } from '../firebaseconfig.js'

export default function RegistrationScreen({ navigation }) {
    const [firstName, setFirstName] = useState('')
    const [lastName, setLastName] = useState('')
    const [foodPreference, setFoodPreference] = useState('')
    const [email, setEmail] = useState('')
    const [password, setPassword] = useState('')
    const [confirmPassword, setConfirmPassword] = useState('')
    const [isCreatingAccount, setIsCreatingAccount] = useState(false)

    const onFooterLinkPress = () => {
        navigation.navigate('Login')
    }
    const onRegisterPress = () => {
        if (password !== confirmPassword) {
            alert('Please provide the correct credentials')
            return
        }

        setIsCreatingAccount(true)

        firebase
            .auth()
            .createUserWithEmailAndPassword(email, password)
            .then(async (response) => {
                console.log('USER FROM REGISTRATION', response.user.uid)

                db.collection('users')
                    .doc(response.user.uid)
                    .set({
                        email: response.user.email,
                        firstName: firstName ? firstName : 'Mysterious',
                        lastName: lastName ? lastName : 'Cook',
                        foodPreference: foodPreference
                            ? foodPreference.toString()
                            : 'vegan',
                        points: 0,
                        favoriteRecipes: {},
                        recipeHistory: {},
                        icon:
                            'http://192.168.1.154:19001/assets/assets/profileIcons/icons8-test-account-100.png?platform=android&hash=64f6306119855c06b5d5fe9e161127bc?platform=android&dev=true&minify=false&hot=false',
                    })
            })
            .catch((error) => {
                alert(error)
            })
            .catch((error) => console.log('ERROR') || alert(error))
            .finally(() => setIsCreatingAccount(false))
    }

    return (
        <View style={styles.container}>
            <KeyboardAwareScrollView
                style={{ flex: 1, width: '100%' }}
                keyboardShouldPersistTaps="always"
            >
                {/* First Name Input */}
                <TextInput
                    style={styles.input}
                    placeholder="First Name"
                    placeholderTextColor="#aaaaaa"
                    onChangeText={(text) => setFirstName(text)}
                    value={firstName}
                    underlineColorAndroid="transparent"
                    autoCapitalize="none"
                />

                {/* Last Name Input */}
                <TextInput
                    style={styles.input}
                    placeholder="Last Name"
                    placeholderTextColor="#aaaaaa"
                    onChangeText={(text) => setLastName(text)}
                    value={lastName}
                    underlineColorAndroid="transparent"
                    autoCapitalize="none"
                />
                <Text
                    style={{
                        marginTop: 5,
                        marginBottom: 5,
                        marginLeft: 25,
                        marginRight: 30,
                        paddingLeft: 10,
                    }}
                >
                    Select your food preference:{' '}
                </Text>

                {/* Preference Picker */}
                <DropDownPicker
                    items={[
                        { label: 'Vegan', value: 'vegan' },
                        { label: 'Meatlover', value: 'meatlover' },
                    ]}
                    style={styles.input}
                    defaultValue="vegan"
                    onChangeItem={(itemValue, itemIndex) => {
                        setFoodPreference(itemValue.value)
                    }}
                ></DropDownPicker>

                {/* Email Input */}
                <TextInput
                    style={styles.input}
                    placeholder="E-mail"
                    placeholderTextColor="#aaaaaa"
                    onChangeText={(text) => setEmail(text)}
                    value={email}
                    underlineColorAndroid="transparent"
                    autoCapitalize="none"
                />

                {/* Password Input */}
                <TextInput
                    style={styles.input}
                    placeholderTextColor="#aaaaaa"
                    secureTextEntry
                    placeholder="Password"
                    onChangeText={(text) => setPassword(text)}
                    value={password}
                    underlineColorAndroid="transparent"
                    autoCapitalize="none"
                />
                <TextInput
                    style={styles.input}
                    placeholderTextColor="#aaaaaa"
                    secureTextEntry
                    placeholder="Confirm Password"
                    onChangeText={(text) => setConfirmPassword(text)}
                    value={confirmPassword}
                    underlineColorAndroid="transparent"
                    autoCapitalize="none"
                />

                {/* Confirm Button */}
                <TouchableOpacity
                    style={styles.button}
                    onPress={onRegisterPress}
                    disabled={isCreatingAccount}
                >
                    {isCreatingAccount ? (
                        <ActivityIndicator size="large"></ActivityIndicator>
                    ) : (
                        <Text style={styles.buttonTitle}>Create account</Text>
                    )}
                </TouchableOpacity>

                <View style={styles.footerView}>
                    <Text style={styles.footerText}>
                        Already got an account?{' '}
                        <Text
                            onPress={onFooterLinkPress}
                            style={styles.footerLink}
                        >
                            Log in
                        </Text>
                    </Text>
                </View>
            </KeyboardAwareScrollView>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        marginTop: 100,
    },
    title: {},
    logo: {
        flex: 1,
        height: 120,
        width: 90,
        alignSelf: 'center',
        margin: 30,
    },
    input: {
        height: 55,
        borderRadius: 5,
        overflow: 'hidden',
        backgroundColor: 'white',
        marginTop: 10,
        marginBottom: 10,
        marginLeft: 30,
        marginRight: 30,
        paddingLeft: 16,
    },
    button: {
        backgroundColor: '#F18F01',
        marginLeft: 30,
        marginRight: 30,
        marginTop: 20,
        height: 48,
        borderRadius: 5,
        alignItems: 'center',
        justifyContent: 'center',
    },
    buttonTitle: {
        color: 'white',
        fontSize: 16,
        fontWeight: 'bold',
    },
    footerView: {
        flex: 1,
        alignItems: 'center',
        marginTop: 20,
    },
    footerText: {
        fontSize: 16,
        color: '#2e2e2d',
    },
    footerLink: {
        color: '#788eec',
        fontWeight: 'bold',
        fontSize: 16,
    },
})
