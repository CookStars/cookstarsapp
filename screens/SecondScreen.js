import React from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';

export default function SecondScreen({ navigation }) {
  return (
    <View style={styles.container}>
      <Text>Second Screen</Text>
      <Button title='Go To Home' onPress={() => navigation.navigate('Home')} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
