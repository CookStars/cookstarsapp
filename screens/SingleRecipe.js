import React, { Component } from 'react'
import HTML from 'react-native-render-html'

import {
    StyleSheet,
    Text,
    ScrollView,
    View,
    Button,
    Image,
    Alert,
    Dimensions,
    TouchableOpacity,
} from 'react-native'
import { Feather, MaterialCommunityIcons } from '@expo/vector-icons'
export default class SingleRecipe extends Component {
    constructor(props) {
        super(props)
    }

    checkDay() {
        const weekdays = [
            'Sunday',
            'Monday',
            'Tuesday',
            'Wednesday',
            'Thursday',
            'Friday',
            'Saturday',
        ]
        const today = weekdays[new Date().getDay()]
        const { navigation } = this.props
        const {
            index,
            recipes,
            userInfo,
            recipeCompleted,
            recipeFinished,
        } = this.props.route.params

        if (weekdays[index] === today && !recipeFinished) {
            return (
                <View style={styles.startButton}>
                    <TouchableOpacity
                        onPress={() => {
                            navigation.navigate('Steps', {
                                index: index,
                                recipes: recipes,
                                userInfo: userInfo,
                                recipeCompleted: recipeCompleted,
                            })
                        }}
                        style={{
                            // elevation: 12,
                            backgroundColor: '#EF233C',
                            borderRadius: 10,
                            width: '100%',
                            height: 35,
                            alignItems: 'center',
                            alignContent: 'center',
                            justifyContent: 'center',
                        }}
                    >
                        <View>
                            <Text style={{ color: 'white' }}>START RECIPE</Text>
                        </View>
                    </TouchableOpacity>
                </View>
            )
        } else {
            return <View />
        }
    }

    render() {
        const { index, recipes } = this.props.route.params
        const {
            summary,
            title,
            image,
            ingredients,
            readyInMinutes,
            servings,
        } = recipes[index]
        const listIngredients = ingredients
            .map((ingredient) => ingredient.original)
            .join(', ')
        const newTagsSummary = summary
            .split(/\<a\b[^>]*>/)
            .join('<b><i>')
            .split(/\<\/a>/)
            .join('</i></b>')

        return (
            <View>
                <View>
                    <ScrollView style={styles.scrollView}>
                        <View style={styles.container}>
                            <View style={styles.image}>
                                <Image
                                    source={{
                                        width: Dimensions.get('screen').width,
                                        height:
                                            0.4 *
                                            Dimensions.get('screen').height,
                                        uri: image,
                                    }}
                                />

                                <TouchableOpacity
                                    onPress={() => {
                                        console.log('favorited')
                                    }}
                                    style={{
                                        backgroundColor: '#EF233C',
                                        position: 'absolute',
                                        top: '85%',
                                        right: -8,
                                        borderRadius: 10,
                                        width: '40%',
                                        height: 35,
                                        alignItems: 'center',
                                        alignSelf: 'center',
                                        justifyContent: 'center',
                                    }}
                                >
                                    <View
                                        style={{
                                            flexDirection: 'row',
                                            alignItems: 'center',
                                        }}
                                    >
                                        <Feather
                                            name="heart"
                                            size={20}
                                            color="white"
                                        />
                                        <Text
                                            style={{ color: 'white', left: 6 }}
                                        >
                                            FAVORITE
                                        </Text>
                                    </View>
                                </TouchableOpacity>
                            </View>
                            <View>
                                <Text style={styles.title}>{title}</Text>
                            </View>
                            <View
                                style={{
                                    flexDirection: 'column',
                                    height: '15%',
                                    justifyContent: 'space-evenly',
                                }}
                            >
                                <View
                                    style={{
                                        flexDirection: 'row',
                                        width:
                                            0.8 *
                                            Dimensions.get('screen').width,
                                        justifyContent: 'space-around',
                                        top: '3%',
                                    }}
                                >
                                    <View
                                        style={{
                                            flexDirection: 'row',
                                            top: '2%',
                                        }}
                                    >
                                        <Feather
                                            name="clock"
                                            size={25}
                                            color="red"
                                            style={{ alignSelf: 'center' }}
                                        />
                                        <Text
                                            style={{
                                                left: '30%',
                                                fontSize: 15,
                                                alignSelf: 'center',
                                            }}
                                        >
                                            {readyInMinutes} minutes
                                        </Text>
                                    </View>
                                    <View
                                        style={{
                                            flexDirection: 'row',
                                            top: '2%',
                                        }}
                                    >
                                        <MaterialCommunityIcons
                                            name="bowl"
                                            size={25}
                                            color="red"
                                        />
                                        <Text
                                            style={{
                                                left: '30%',
                                                fontSize: 15,
                                                alignSelf: 'center',
                                            }}
                                        >
                                            {servings} Servings
                                        </Text>
                                    </View>
                                </View>
                            </View>
                        </View>

                        <View
                            style={{
                                flexDirection: 'column',
                                justifyContent: 'space-between',
                                paddingBottom: 130,
                                alignItems: 'center',
                                flex: 6,
                            }}
                        >
                            <View style={styles.description}>
                                <Text
                                    style={{
                                        fontWeight: 'bold',
                                        fontSize: 18,
                                        height: '10%',
                                    }}
                                >
                                    Description:
                                </Text>
                                <HTML
                                    html={newTagsSummary}
                                    baseFontStyle={styles.description}
                                />
                            </View>
                            <View>
                                <Text
                                    style={{
                                        fontWeight: 'bold',
                                        fontSize: 18,
                                        height: 35,
                                        top: '10%',
                                    }}
                                >
                                    Ingredients:
                                </Text>
                                <Text style={styles.ingredients}>
                                    {listIngredients}
                                </Text>
                            </View>
                        </View>
                    </ScrollView>
                </View>
                <View>
                    <View>{this.checkDay()}</View>
                </View>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    scrollView: {
        backgroundColor: 'white',
    },
    container: {
        flex: 10,
        width: Dimensions.get('screen').width,
        height: 0.6 * Dimensions.get('screen').height,
        backgroundColor: 'white',
        alignContent: 'space-between',
    },
    title: {
        left: '6%',
        width: 0.9 * Dimensions.get('screen').width,
        fontSize: 28,
        marginTop: '2%',
        fontWeight: 'bold',
        color: 'black',
    },
    image: {
        borderColor: '#EBEBD3',
        alignItems: 'center',
        resizeMode: 'stretch',
    },
    description: {
        borderColor: '#F78764',
        top: '2%',
        alignSelf: 'center',
        width: 0.88 * Dimensions.get('screen').width,
        fontSize: 15,
        textAlign: 'auto',
    },
    ingredients: {
        top: '10%',
        padding: 3,
        borderWidth: 3,
        borderColor: 'silver',
        alignSelf: 'center',
        width: 0.88 * Dimensions.get('screen').width,
        fontSize: 15,
        textAlign: 'auto',
    },

    startButton: {
        width: '50%',
        height: 40,
        justifyContent: 'center',
        alignSelf: 'center',
        position: 'absolute',
        bottom: 0,
    },
})
