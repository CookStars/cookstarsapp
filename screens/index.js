import SecondScreen from "./SecondScreen";
import Home from "./Home";
import Login from "./Login";

export { SecondScreen, Home, Login };
