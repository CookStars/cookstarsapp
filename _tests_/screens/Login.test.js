import React from 'react'
import { configure, shallow } from 'enzyme'
import { Login } from '../../screens/index'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
// import { ThemeProvider } from 'react-native'

describe('Test case for testing login', () => {
    configure({ adapter: new Adapter() })

    it('renders correctly', () => {
        const tree = renderer.create(<Login />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('checks email input', () => {
        const wrapper = shallow(<Login />)
        expect(wrapper.find('#emailInput').length).toEqual(1)
    })

    it('checks email field sets state on change', () => {
        const wrapper = shallow(<Login />)
        wrapper.find('#emailInput').simulate('changeText', 'email@email.com')
        // console.log('@@@@value', wrapper.find('#emailInput').props().value)
        expect(wrapper.state('email')).toEqual('email@email.com')
    })
})
