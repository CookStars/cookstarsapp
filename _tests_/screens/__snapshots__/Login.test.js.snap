// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for testing login renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F4F1DE",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 20,
        "marginTop": 10,
      }
    }
  />
  <View />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#F2CC8F",
        "borderBottomWidth": 1,
        "borderRadius": 30,
        "flexDirection": "row",
        "height": 45,
        "marginBottom": 20,
        "width": 250,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      id="emailInput"
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="Email"
      rejectResponderTermination={true}
      style={
        Object {
          "borderBottomColor": "#F2CC8F",
          "flex": 1,
          "height": 45,
          "marginLeft": 16,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#F2CC8F",
        "borderBottomWidth": 1,
        "borderRadius": 30,
        "flexDirection": "row",
        "height": 45,
        "marginBottom": 20,
        "width": 250,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "borderBottomColor": "#F2CC8F",
          "flex": 1,
          "height": 45,
          "marginLeft": 16,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 30,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 20,
          "width": 250,
        },
        Object {
          "backgroundColor": "#F18F01",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      Login
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 30,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginBottom": 20,
        "width": 250,
      }
    }
  >
    <Text>
      Forgot your password?
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 30,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginBottom": 20,
        "width": 250,
      }
    }
  >
    <Text>
      Register
    </Text>
  </View>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
